### PR TITLE
support avro union type in sources

### LINKED ISSUE #2359: Document: feat(source): support Avro Union type
This issue tracks the documentation update needed for the merged PR #17485.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/17485
Source PR Merged At: 2024-07-09T12:08:46Z

### PR DESCRIPTION
**Pull request by ShanlanLi**
## Description
image

## Related code PR

https://github.com/risingwavelabs/risingwave/pull/17485/files

## Related doc issue

Resolve https://github.com/risingwavelabs/risingwave-docs/issues/2359

## Checklist

- [ ] I have checked the doc site preview, and the updated parts look good.
- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`emile-00`, `hengm3467`, & `WanYixian`).

### CODE REVIEW

**Review by xxchan — COMMENTED** [review]
I think it might be better to doc it using the way in #2294

i.e., update this line:
image

File: docs/ingest/supported-sources-and-formats.md [hunk 1]
```diff
@@ -82,6 +82,26 @@ FORMAT [ DEBEZIUM | UPSERT | PLAIN ] ENCODE AVRO (
 
 Note that the value types can only be: `null`, `boolean`, `int`, `string`, or `map`/`record`/`array` with these types.
 
+From version 2.0, Risingwave supports Avro Union type for sources. For example, let's assume you have created a Kafka topic `avro-union-simple` and registered the Avro schema with fields containing a Union type (such as ["int", "string", "null", "boolean"]). After producing messages to this topic, you can create a source and query the data as shown below:
+
+```sql
+CREATE SOURCE avro_union
+WITH (
+      ${RISEDEV_KAFKA_WITH_OPTIONS_COMMON},
```
Comment (xxchan) [inline docs/ingest/supported-sources-and-formats.md:90]: Just FYI: `${RISEDEV_KAFKA_WITH_OPTIONS_COMMON}` is not for users. It's a development only thing. We should respect the example style like other places in the doc.

**Comment by ShanlanLi** [comment]
> I think it might be better to doc it using the way in #2294
> 
> i.e., update this line: image

Thanks for your kind suggestion and reminding. Encoding and format are a bit hard for me to understand. That's why mistakes happen. Am trying to study more about them.

As for the page(WIP: refactor(connector): consolidate format & encode options #2294) you mentioned, I worry it might not be available for 2.0 doc. Can we have a replacement plan to update based on the 1.10 doc?

> I think it might be better to doc it using the way in #2294
image

**Comment by xxchan** [comment]
> I worry it might not be available for 2.0 doc. Can we have a replacement plan to update based on the 1.10 doc?

I think we can just simply do not doc it now. Because this is more like a "reference doc" to me. Users don't need to anything special to use this feature (and thus won't read the doc), and is decided by their data.

**Comment by ShanlanLi** [comment]
Will be documented in the future

**Pull request closed without merging**